internal/security/fileprovider/provider.go

```diff
@@ -153,22 +153,20 @@ func (p *fileTokenProvider) Run() error {
 			p.logger.Error(fmt.Sprintf("failed open token file for writing %s: %s", outputTokenFilename, err.Error()))
 			return err
 		}
-		defer func() {
-			if writeCloser != nil {
-				writeCloser.Close()
-			}
-		}()
+		// writeCloser is writable file -- explicitly close() to ensure we catch errors writing to it
 
 		permissionable, ok := writeCloser.(permissionable)
 		if ok {
 			if serviceConfig.FilePermissions != nil &&
 				(serviceConfig.FilePermissions).ModeOctal != nil {
 				mode, err := strconv.ParseInt(*(serviceConfig.FilePermissions).ModeOctal, 8, 32)
 				if err != nil {
+					writeCloser.Close()
```

Comment:
Recommend explicitly swallowing the error return value; i.e. `_ = writeCloser.Close()`

Mentioned once, occurs elsewhere.

Reply:
Implemented as suggested 5X.